pkg/chunk/inmemory_storage_client.go

```diff
@@ -311,14 +332,82 @@ func (m *MockStorage) GetChunks(ctx context.Context, chunkSet []Chunk) ([]Chunk,
 	return result, nil
 }
 
-type mockWriteBatch []struct {
-	tableName, hashValue string
-	rangeValue           []byte
-	value                []byte
+// DeleteChunk implements StorageClient.
+func (m *MockStorage) DeleteChunk(ctx context.Context, chunkID string) error {
```

Comment:
I'm a bit confused. Could you explain me why we need to add these functions (`DeleteChunk()`, `GetObject()`, etc) to the `MockStorage` please?

Reply:
`MockStorage` is used at various places for doing store tests. I think for this PR I don't need them. I will need it for my next PR so I can either leave it here for now or add it in the next PR. What do you think?

Reply:
I'm OK keeping it if necessary (whether in this or subsequent PR). The only thing which sounds as a bad smell to me (but I may be wrong cause I didn't check all usages of `MockStorage`) is that after Jacob refactoring we now have two interfaces: one for the chunk client and one for the object store client. Here we're mixing both interfaces in the same implementation and I was wondering if this is required or not. If yes, I'm good.

Reply:
`MockStorage` is jack of all trades for tests that need stores. Which means it needs to satisfy both the interfaces.
Regarding reusing code, there is a package called `objectclient` which takes in `chunks.ObjectClient` and converts it to `objectclient.Client`. `chunks.ObjectClient` works on raw objects while the methods in `objectclient.Client` reuse that code to allow working on chunks. The package `objectclient` imports `chunks` package while the implementation of `MockStorage` lives in `chunks` package which means I can't import `objectclient` back in `chunks` package to reuse the code which does the conversion of interfaces.